lib/object.c

```diff
@@ -196,21 +196,36 @@ static tool_rc tpm2_util_object_fetch_parent_from_tpk(const char *objectstr,
     tool_rc rc = tool_rc_general_error;
     BIO *input_bio = 0;
     TSSPRIVKEY_OBJ *tpk = 0;
+    BIGNUM *bn_parent = 0;
     tpm2_util_object_fetch_tssprivkey_from_file(objectstr, &input_bio, &tpk);
     if (!input_bio || !tpk) {
         goto ret;
     }
 
-    uint64_t parent;
-    int ret = ASN1_INTEGER_get_uint64(&parent, tpk->parent);
-    if (ret == 0 || parent > UINT32_MAX) {
+    bn_parent = ASN1_INTEGER_to_BN(tpk->parent, NULL);
+    if (!bn_parent) {
+        goto ret;
+    }
+
+    TPM2_HANDLE parent = 0;
+    bool is_bn_parent_negative = BN_is_negative(bn_parent);
+    if (is_bn_parent_negative) {
+        parent = ASN1_INTEGER_get(tpk->parent);
```

Comment:
I guess this is how the other code does it, wonder how it's going to fair with the deprecation mention, but the header files have yet to tag it with a deprecated attribute to cause a warning AFAICT.